src/App.js

```diff
@@ -1,16 +1,52 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import './App.css';
+
+const sender1 = chatMessages[0].sender;
+const sender2 = chatMessages[1].sender;
 
 const App = () => {
+  const [entries, setEntries] = useState(chatMessages);
+  const updateChat = (id) => {
+    const newEntries = entries.map((entry) => {
+      if (entry.id === id) {
+        // same as below:
+        // return Object.assign({}, entry, { liked: !entry.liked });
+        return { ...entry, liked: !entry.liked };
+      }
+      return entry;
+    });
+    setEntries(newEntries);
+  };
+
+  const calcTotalLikes = (entries) => {
+    return entries.reduce((total, entry) => {
+      return total + entry.liked;
```

Comment:
Great way to do this calculation! Great use of the boolean unary operator! Nice use of reduce to tally up the liked message count from state we were already tracking, without the need for an additional piece of state.